irohad/consensus/yac/impl/yac_gate_impl.cpp

```diff
@@ -35,103 +35,101 @@ namespace iroha {
             block_loader_(std::move(block_loader)),
             consensus_result_cache_(std::move(consensus_result_cache)),
             log_(logger::log("YacGate")) {
-        block_creator_->on_block().subscribe(
-            [this](auto block) { this->vote(block); });
+        block_creator_->on_block().subscribe([this](auto block) {
+          // TODO(@l4l) 24/09/18 IR-1717
+          // update BlockCreator iface according to YacGate
+          this->vote({std::shared_ptr<shared_model::interface::Proposal>()},
+                     {block},
+                     {0, 0});
+        });
       }
 
-      void YacGateImpl::vote(std::shared_ptr<shared_model::interface::Block> block) {
-        auto hash = hash_provider_->makeHash(*block);
-        log_->info("vote for block ({}, {})",
-                   hash.vote_hashes.proposal_hash,
-                   block->hash().toString());
-        auto order = orderer_->getOrdering(hash);
+      void YacGateImpl::vote(
+          boost::optional<std::shared_ptr<shared_model::interface::Proposal>>
+              proposal,
+          boost::optional<std::shared_ptr<shared_model::interface::Block>>
+              block,
+          Round round) {
+        bool is_none = not proposal or not block;
+        if (is_none) {
+          current_block_ = nullptr;
+          current_hash_ = {};
+          log_->debug("Agreed on nothing to commit");
+        } else {
+          current_block_ = std::move(block.value());
+          current_hash_ =
+              hash_provider_->makeHash(*current_block_ /*, *proposal, round*/);
+          log_->info("vote for (proposal: {}, block: {})",
+                     current_hash_.vote_hashes.proposal_hash,
+                     current_hash_.vote_hashes.block_hash);
+        }
+
+        auto order = orderer_->getOrdering(current_hash_);
         if (not order) {
           log_->error("ordering doesn't provide peers => pass round");
           return;
         }
-        current_block_ = std::make_pair(hash, block);
-        hash_gate_->vote(hash, *order);
+
+        hash_gate_->vote(current_hash_, *order);
 
         // insert the block we voted for to the consensus cache
-        consensus_result_cache_->insert(block);
+        if (is_none) {
```

Comment:
Code in this conditional might throw an exception in case there was no block:
https://github.com/hyperledger/iroha/blob/a52ca3e89997bcb45e04d3032dc8258da436673d/irohad/consensus/yac/impl/yac_gate_impl.cpp#L53
Check if this case is covered by the tests and change the conditional.